### PR TITLE
Docs: fix 10 out of 16 flow fixme's

### PR DESCRIPTION
**Pull request by christianvuerings**
#### What changed?

Fixes 10 out of 16 flow fixme in `/docs/`

#### Why?

Improve flow coverage

### CODE REVIEW

File: docs/docs-components/Overview.js [hunk 1]
```diff
@@ -32,14 +32,13 @@ export default function Overview({
...
-    (acc, cur) => ({
-      ...acc,
-      // $FlowFixMe[invalid-computed-prop]
-      [cur.category]: (acc[cur.category] ?? []).concat(cur),
-    }),
-    {},
-  );
+  const GENERAL_COMPONENT_CATEGORY_MAP = generalComponents.reduce((acc, cur) => {
+    const { category } = cur;
+    if (category) {
```
Comment (christianvuerings) [inline docs/docs-components/Overview.js:37]: fyi: https://github.com/facebook/flow/issues/8299

File: docs/docs-components/Header.js [hunk 2]
```diff
@@ -35,8 +43,6 @@ function SettingsDropdown({
   };
   return (
     <Dropdown
-      // $FlowFixMe[cannot-read] Flow is wrong here
```
Comment (AlbertCarreras) [inline docs/docs-components/Header.js:38]: nice

File: docs/utils/mdHelper.js [hunk 3]
```diff
@@ -45,12 +45,12 @@ export async function getAllMarkdownPosts(): Promise<$ReadOnlyArray<$ReadOnlyArr
     ) => {
       // for each choice
       pageItems.forEach((page) => {
-        if (page.sectionName) {
-          // $FlowFixMe[incompatible-type] This is a siteIndexType because it has a section name and isn't a string
-          const siteIndexSection: siteIndexType = page;
+        if (typeof page !== 'string') {
```
[the remark this answers is not in the document]
Comment (rlingineni) [reply docs/utils/mdHelper.js:48]: Ah got it! This is easier. I was trying to find an `as` modifier to cast like typescript,  but it didn't exist.

I guess once we can guarantee it's an object, then we'd also be allowed to cast it to any object type

**Review by dangerismycat — APPROVED** [review]
Thank you Christian! ❤️

**Pull request merged**